clang/lib/Driver/Driver.cpp

```diff
@@ -716,7 +716,6 @@ void Driver::CreateOffloadingDeviceToolChains(Compilation &C,
     C.addOffloadDeviceToolChain(CudaTC, OFK);
   } else if (IsHIP) {
     const ToolChain *HostTC = C.getSingleOffloadToolChain<Action::OFK_Host>();
-    const llvm::Triple &HostTriple = HostTC->getTriple();
```

Comment:
It seems like that this warning was introduced by community code. We do regular pulldowns from llvm project. Should this patch go through llorg instead?
See git blame:

```
f614422da9bdd (Yaxun Liu               2018-05-30 00:53:50 +0000  717)   } else if (IsHIP) {
f614422da9bdd (Yaxun Liu               2018-05-30 00:53:50 +0000  718)     const ToolChain *HostTC = C.getSingleOffloadToolChain<Action::OFK_Host>();
f614422da9bdd (Yaxun Liu               2018-05-30 00:53:50 +0000  719)     const llvm::Triple &HostTriple = HostTC->getTriple();
f614422da9bdd (Yaxun Liu               2018-05-30 00:53:50 +0000  720)     auto OFK = Action::OFK_HIP;
7546b29e76168 (Yaxun (Sam) Liu         2020-05-04 23:22:06 -0400  721)     llvm::Triple HIPTriple = getHIPOffloadTargetTriple();
25cd3108011b5 (Andrew Gozillon         2019-05-28 19:37:04 -0700  722)     // Use the HIP and host triples as the key into
25cd3108011b5 (Andrew Gozillon         2019-05-28 19:37:04 -0700  723)     // getOffloadingDeviceToolChain, because the device toolchain we create
25cd3108011b5 (Andrew Gozillon         2019-05-28 19:37:04 -0700  724)     // depends on both.
25cd3108011b5 (Andrew Gozillon         2019-05-28 19:37:04 -0700  725)     auto HIPTC = &getOffloadingDeviceToolChain(C.getInputArgs(), HIPTriple,
25cd3108011b5 (Andrew Gozillon         2019-05-28 19:37:04 -0700  726)                                                *HostTC, OFK);
25cd3108011b5 (Andrew Gozillon         2019-05-28 19:37:04 -0700  727)     C.addOffloadDeviceToolChain(HIPTC, OFK);
c1ffba5062c7f (Samuel Antao            2016-06-13 18:10:57 +0000  728)   }

```

Reply:
> It seems like that this warning was introduced by community code.

I don't think so. @Fznamznon, @AGindinson, take a closer look at [this patch](https://github.com/intel/llvm/commit/25cd3108011b5d665cb0679112c134f082c5d62b#diff-f8c64e36dfe3828a6f816c4550e78bb0305769ace1be53207e86ac9a3280ac9eR631-R642).

Reply:
Ah, ok. I was wrong. Thanks!

Reply:
Ah, I think I get it: a patch @bader has swiftly linked had removed the use of the variable specifically for intel/llvm. Therefore, removing the variable itself should be considered as a mere follow-up to that change. If such reasoning is valid, the intel/llvm-specific refactoring should be welcome.